source/common/ssl/ssl_socket.cc

```diff
@@ -290,6 +308,23 @@ std::string SslSocket::getUriSanFromCertificate(X509* cert) {
   return "";
 }
 
+std::vector<std::string> SslSocket::getDnsSansFromCertificate(X509* cert) {
+  bssl::UniquePtr<GENERAL_NAMES> san_names(
+      static_cast<GENERAL_NAMES*>(X509_get_ext_d2i(cert, NID_subject_alt_name, nullptr, nullptr)));
+  if (san_names == nullptr) {
+    return {};
+  }
+  std::vector<std::string> dns_sans = {};
+  for (const GENERAL_NAME* san : san_names.get()) {
+    if (san->type == GEN_DNS) {
+      ASN1_STRING* dns = san->d.dNSName;
+      dns_sans.push_back(std::string(reinterpret_cast<const char*>(ASN1_STRING_data(dns)),
```

Comment:
Per my previous comment, I think this could be:

```
dns_sans.emplace_back(reinterpret_cast<const char*>(ASN1_STRING_data(dns)),
                      ASN1_STRING_length(dns));
```

Reply:
The duplicate `#include` removal got lost in the rebase I think, made the changes.